Makefile

```diff
@@ -15,205 +15,182 @@
-
-ifneq ($(TARGET_NAME),TARGET_STAX)
-SDK_SOURCE_PATH  += lib_ux
+# restricted features for Nano S
+# but not in DEVEL mode where we usually want to test all features with HEADLESS
+ifeq ($(TARGET_NAME),TARGET_NANOS)
+	ifneq ($(DEVEL), 1)
+		APP_XS = 1
+	endif
+else
+	APP_XS = 0
 endif
 
-ifeq ($(TARGET_NAME),$(filter $(TARGET_NAME),TARGET_NANOX TARGET_STAX))
-	SDK_SOURCE_PATH  += lib_blewbxx lib_blewbxx_impl
+ifeq ($(APP_XS), 1)
+	DEFINES += APP_XS
+else
+	# features not included in the Nano S app
+	DEFINES += APP_FEATURE_OPCERT
+	DEFINES += APP_FEATURE_NATIVE_SCRIPT_HASH
+	DEFINES += APP_FEATURE_POOL_REGISTRATION
+	DEFINES += APP_FEATURE_POOL_RETIREMENT
+	DEFINES += APP_FEATURE_BYRON_ADDRESS_DERIVATION
+	DEFINES += APP_FEATURE_BYRON_PROTOCOL_MAGIC_CHECK
 endif
+# always include this, it's important for Plutus users
+DEFINES += APP_FEATURE_TOKEN_MINTING
 
-################
-# Default rule #
-################
-
-all: default
-
-
-##############
-#   Build    #
-##############
-
-# import generic rules from the sdk
-include $(BOLOS_SDK)/Makefile.rules
-
-#add dependency on custom makefile filename
-dep/%.d: %.c Makefile
-
-listvariants:
-	@echo VARIANTS COIN cardano_ada
-
-# part of CI
-analyze: clean
-	scan-build --use-cc=clang -analyze-headers -enable-checker security -enable-checker unix -enable-checker valist -o scan-build --status-bugs make default
-
-##############
-#   Load     #
-##############
+########################################
+#     Application custom permissions   #
+########################################
+# See SDK `include/appflags.h` for the purpose of each permission
+#HAVE_APPLICATION_FLAG_DERIVE_MASTER = 1
+#HAVE_APPLICATION_FLAG_GLOBAL_PIN = 1
+HAVE_APPLICATION_FLAG_BOLOS_SETTINGS = 1
+#HAVE_APPLICATION_FLAG_LIBRARY = 1
+
+########################################
+# Application communication interfaces #
+########################################
+ENABLE_BLUETOOTH = 1
+#ENABLE_NFC = 1
+
+########################################
+#         NBGL custom features         #
+########################################
+ENABLE_NBGL_QRCODE = 1
+#ENABLE_NBGL_KEYBOARD = 1
+#ENABLE_NBGL_KEYPAD = 1
+
+########################################
+#          Features disablers          #
+########################################
+# These advanced settings allow to disable some feature that are by
+# default enabled in the SDK `Makefile.standard_app`.
+#DISABLE_STANDARD_APP_FILES = 1
+#DISABLE_DEFAULT_IO_SEPROXY_BUFFER_SIZE = 1 # To allow custom size declaration
+#DISABLE_STANDARD_APP_DEFINES = 1 # Will set all the following disablers
+#DISABLE_STANDARD_SNPRINTF = 1
+#DISABLE_STANDARD_USB = 1
+#DISABLE_STANDARD_WEBUSB = 1
+#DISABLE_STANDARD_BAGL_UX_FLOW = 1
+
+########################################
+#       Additional configuration       #
+########################################
+
+# USB U2F
+DEFINES += HAVE_U2F HAVE_IO_U2F U2F_PROXY_MAGIC=\"ADA\"
+SDK_SOURCE_PATH  += lib_u2f
+
+DEFINES += HAVE_UX_LEGACY
```

Comment:
This wasn't included in the previous version, is it really needed?

Reply:
I think what you need is something like:
```
ifeq ($(TARGET_NAME), TARGET_NANOS)
    DISABLE_STANDARD_BAGL_UX_FLOW=1
endif
```
To keep the same kind of stuff that you had before.
Though dropping the usage of old nanos ux code, to use the common one would be an interssting improvement.
I see that it's automatically done if you have HAVE_FLOW in uiHelpers_nanos.c and uiHelpers_nanox.c but not in menu_nanos.c, menu_nanox.c, menu.h, io.c, uiHelpers.h, io.h, uiScreens_bagl.c, uiHelpers.c

Reply:
I've tried it, however, I'd say that `lib_standard_app/main.c` is not compatible with it:
```
/opt/nanos-secure-sdk/lib_standard_app/main.c:33:19: error: redefinition of 'ux' with a different type: 'bolos_ux_params_t' (aka 'struct bolos_ux_params_s') vs 'ux_state_t' (aka 'struct ux_state_s')
bolos_ux_params_t G_ux_params;
                  ^
/opt/nanos-secure-sdk/lib_ux/include/ux.h:322:30: note: expanded from macro 'G_ux_params'
#define G_ux_params          ux.params
                             ^
/opt/nanos-secure-sdk/lib_standard_app/main.c:32:19: note: previous definition is here
ux_state_t        G_ux;
                  ^
/opt/nanos-secure-sdk/lib_ux/include/ux.h:321:30: note: expanded from macro 'G_ux'
#define G_ux                 ux
                             ^
/opt/nanos-secure-sdk/lib_standard_app/main.c:33:19: error: expected ';' after top level declarator
bolos_ux_params_t G_ux_params;
                  ^
/opt/nanos-secure-sdk/lib_ux/include/ux.h:322:32: note: expanded from macro 'G_ux_params'
#define G_ux_params          ux.params
```
@xchapron-ledger Any advice, please?

Reply:
(At some point, we might rewrite the UI to use the new version, but at present we don't have time and budget for that.)

Reply:
As you want, I'm even wondering if you have anything to do.
I looks like you could simply:
- Enable HAVE_FLOW for TARGET_NANOS
- Update some #ifdef in menu_nanos.c, menu_nanox.c, menu.h, io.c, uiHelpers.h, io.h, uiScreens_bagl.c, uiHelpers.c
And that should be all?
But could be for another time.